File: basic/minimum_path/problem1.py
N, M = map(int, input().split())
dp = [[int(1e9)]*(N+1) for _ in range(N+1)]

for i in range(N+1):
    for j in range(N+1):
        if i == j: dp[i][j] = 0

for _ in range(M):
    i, j = map(int, input().split())
    dp[i][j] = 1
    dp[j][i] = 1

X, K = map(int, input().split())

for i in range(1, N+1):
    for j in range(1, N+1):
        for k in range(1, N+1):
            dp[j][k] = min(dp[j][k], dp[j][i] + dp[i][k])

sol = dp[1][K] + dp[K][X]
if  sol > int(1e9): print(-1)
else: print(sol)

File: basic/binary_search/problem1.cpp
from bisect import bisect_left, bisect_right

N, x = map(int, input().split())

sequence = list(map(int, input().split()))

x_count = bisect_right(sequence, x) - bisect_left(sequence, x)
if x_count == 0 : x_count = -1
print(x_count)